prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt

```diff
@@ -141,4 +155,88 @@ class DatabaseDeliveryAccess(
             query.fetchInto(DeliveryFacility::class.java)
         }
     }
+
+    /**
+     * Get a list of matching delivery facilities for reports delivered to a given [receiver]
+     * @todo further work on https://github.com/CDCgov/prime-reportstream/issues/9272
+     * @param receiver the organization that is being checked.
+     * @param receivingOrgSvc is a specifier for the receiving organization's service.
+     * @param sortDir sort the table by date in ASC or DESC order
+     * @param sortColumns sort the table by a specific set of columns
+     * @param since is the OffsetDateTime minimum date to get results for.
+     * @param until is the OffsetDateTime maximum date to get results for.
+     * @return List of matching delivery facilities
+     */
+    fun fetchBulkDeliveryFacilities(
+        receiver: String,
+        receivingOrgSvc: String?,
+        sortDir: SortDir,
+        sortColumns: List<BulkFacilitySortColumn>,
+        since: OffsetDateTime?,
+        until: OffsetDateTime?,
+    ): List<DeliveryFacility> {
+        val reportIds = db.transactReturning { txn ->
+            var filter = this.organizationFilter(receiver, receivingOrgSvc)
```

Comment:
Taking another look at this, I think this might not be specific enough of a filter as there are reports that will come back that are the report before it actually get's delivered right?

https://github.com/CDCgov/prime-reportstream/blob/06da55ad45269c893fdbd05d8a4b22763b5f62db/prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt#L50